src/service/mercury/index.ts

```diff
@@ -0,0 +1,162 @@
+import { Client } from "@urql/core";
+import axios from "axios";
+import { Logger } from "pino";
+import { nativeToScVal } from "soroban-client";
+import { mutation, query } from "./queries";
+
+export interface NewSubscriptionPayload {
+  contract_id?: string;
+  max_single_size: number;
+  [key: string]: string | number | undefined;
+}
+
+interface MercurySession {
+  baseUrl: string;
+  email: string;
+  password: string;
+  token: string;
+  userId: string;
+}
+
+export class MercuryClient {
+  urqlClient: Client;
+  mercurySession: MercurySession;
+  mercuryNewSubUrl: string;
+  logger: Logger;
+
+  constructor(
+    mercurySession: MercurySession,
+    urqlClient: Client,
+    logger: Logger
+  ) {
+    this.mercurySession = mercurySession;
+    this.mercuryNewSubUrl = `${mercurySession.baseUrl}:3030/newsubscription`;
```

Comment:
is it worth making the port number configurable? Might this change depending on the `baseUrl`?

Reply:
They've been consistent so far but easy enough to pull these out into config. I'll do that.

Reply:
![image](https://github.com/user-attachments/assets/dcfd3518-521d-46e2-91ec-931d3d992f5c)